src/schema.cpp

```diff
@@ -77,6 +78,66 @@ Schema::const_iterator Schema::find(ObjectSchema const& object) const noexcept
     return const_cast<Schema *>(this)->find(object);
 }
 
+namespace {
+
+struct CheckObjectPath {
+    const ObjectSchema& object;
+    std::string path;
+};
+
+struct CheckObjectPathHash {
+    size_t operator()(CheckObjectPath const& o) const
+    {
+        return std::hash<std::string>{}(o.object.name);
+    }
+};
+
+inline bool operator==(const CheckObjectPath& lhs, const CheckObjectPath& rhs)
+{
+    return lhs.object.name == rhs.object.name;
+}
+
+// a non-recursive search that returns a path to any embedded object that has multiple paths from the start
+std::string do_check(Schema const& schema, const ObjectSchema& start)
+{
+    std::unordered_set<CheckObjectPath, CheckObjectPathHash> to_visit;
```

Comment:
This could now be just `std::queue<CheckObjectPath>` since there's no lookups in it.